### PR TITLE
generate-paths-for-conditional-resources

### PR DESCRIPTION
**Pull request by LubomirS**
added environment for path generation

### CODE REVIEW

File: schema-generator/src/main/java/com/github/jrcodeza/schema/generator/OpenAPIGenerator.java [hunk 1]
```diff
@@ -196,7 +194,12 @@ private Paths createPathsWrapper() {
     }
 
     private Map<String, PathItem> createPathExtensions() {
-        ClassPathScanningCandidateComponentProvider scanner = new ClassPathScanningCandidateComponentProvider(false);
+		ClassPathScanningCandidateComponentProvider scanner;
```
Comment (jrcodeza) [inline schema-generator/src/main/java/com/github/jrcodeza/schema/generator/OpenAPIGenerator.java:197]: Create method for this so we don't have if else structure for initialization inside a method that does other things based on name

File: schema-v2-generator/src/main/java/com/github/jrcodeza/schema/v2/generator/OpenAPIV2Generator.java [hunk 2]
```diff
@@ -127,31 +124,36 @@ public Swagger generate() {
...
 	}
 
 	public void addGlobalHeader(String name, String description, boolean required) {
 		globalHeaders.add(new Header(name, description, required));
 	}
 
 	private Map<String, Path> createPaths(OpenApiV2GeneratorConfig config) {
-		ClassPathScanningCandidateComponentProvider scanner = new ClassPathScanningCandidateComponentProvider(false);
+		ClassPathScanningCandidateComponentProvider scanner;
+		if (environment == null) {
```
Comment (jrcodeza) [inline schema-v2-generator/src/main/java/com/github/jrcodeza/schema/v2/generator/OpenAPIV2Generator.java:152]: same as for v3

File: schema-generator/src/main/java/com/github/jrcodeza/schema/generator/OpenAPIGenerator.java (hunk 1 above)
Comment (LubomirS) [reply schema-generator/src/main/java/com/github/jrcodeza/schema/generator/OpenAPIGenerator.java]: done, thanks

File: schema-v2-generator/src/main/java/com/github/jrcodeza/schema/v2/generator/OpenAPIV2Generator.java (hunk 2 above)
Comment (LubomirS) [reply schema-v2-generator/src/main/java/com/github/jrcodeza/schema/v2/generator/OpenAPIV2Generator.java]: done

**Pull request merged**